src/subs/WCSFitRoutine.cpp

```diff
@@ -382,148 +382,145 @@ void WCSFit::setObjects(int i, const map<std::string, std::vector<double>> &tabl
+        // Note that fitOnce() remaps *all* the matches, including reserved ones.
+        double max;
+        int dof;
+        ca.chisqDOF(dof, max, false);                    // Exclude reserved Matches
+        double thresh = sqrt(chisq / dof) * clipThresh;  // ??? change dof to expectedChisq?
+        if (verbose >= 1)
+            std::cerr << "After iteration: chisq " << chisq << " / "
+                        << dof << " dof, max deviation " << max
+                        << "  new clip threshold at: " << thresh << " sigma" << std::endl;
+        if (thresh >= max || (oldthresh > 0. && (1 - thresh / oldthresh) < minimumImprovement)) {
+            // Sigma clipping is no longer doing much.  Quit if we are at full precision,
+            // else require full target precision and initiate another pass.
+            if (coarsePasses) {
                 coarsePasses = false;
                 ca.setRelTolerance(chisqTolerance);
                 PROGRESS(1, Starting strict tolerance passes);
                 if (clipEntireMatch && verbose >= 1) std::cerr << "-->clipping full matches" << std::endl;
+                oldthresh = thresh;
+                nclip = ca.sigmaClip(thresh, false, clipEntireMatch && !coarsePasses, verbose >= 1);
+                if (verbose >= 0)
+                    std::cerr << "# Clipped " << nclip << " matches " << std::endl;
                 continue;
+            } else {
+                // Done!
+                break;
             }
-            if (verbose >= 0) std::cerr << "# Clipped " << nclip << " matches " << std::endl;
-
-        } while (coarsePasses || nclip > 0);
-
-        // If there are reserved Matches, run sigma-clipping on them now.
-        if (reserveFraction > 0.) {
-            PROGRESS(1, Clipping reserved matches);
-            clipReserved<Astro>(ca, clipThresh, minimumImprovement, false, verbose >= 1);
         }
+        oldthresh = thresh;
+        // Clip entire matches on final passes if clipEntireMatch=true
+        nclip = ca.sigmaClip(thresh, false, clipEntireMatch && !coarsePasses, verbose >= 1);
+        if (nclip == 0 && coarsePasses) {
+            // Nothing being clipped; tighten tolerances and re-fit
+            coarsePasses = false;
+            ca.setRelTolerance(chisqTolerance);
+            PROGRESS(1, Starting strict tolerance passes);
+            if (clipEntireMatch && verbose >= 1) std::cerr << "-->clipping full matches" << std::endl;
+            continue;
+        }
+        if (verbose >= 0) std::cerr << "# Clipped " << nclip << " matches " << std::endl;
 
-        //////////////////////////////////////
-        // Output data and calculate some statistics
-        //////////////////////////////////////
+    } while (coarsePasses || nclip > 0);
 
-        // Report summary of residuals to stdout
-        Astro::reportStatistics(matches, exposures, extensions, cout);
-    } catch (std::runtime_error &m) {
-        quit(m, 1);
```

Comment:
So this was the thing causing the problems?  It was catching all errors and then quitting out?  😬

Reply:
Yeah, it's partially my fault as I left this in during the big refactoring...